### PR TITLE
feat: allow setting vm firmware

### PR DESCRIPTION
**Pull request by ghouscht**
Hi,
As stated in https://github.com/terra-farm/go-virtualbox/pull/28  I need this to create VMs which only support EFI boot, which is now working with these changes.
Thank you,
Thomas

### CODE REVIEW

**Review by VoyTechnology — CHANGES_REQUESTED** [review]
The changes you added look good, but what is missing is adding the ability to read the firmware back in ` resourceVMRead` function.

```go
if err = d.Set("firmware", vm.Firmware); err != nil {
	return diag.Errorf("can't set firmware: %v", err)
}
```

File: virtualbox/resource_vm.go [hunk 1]
```diff
@@ -80,6 +80,13 @@ func resourceVM() *schema.Resource {
 				Default:  "512mib",
 			},
 
+			"firmware": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Firmware of the vm, can either be BIOS or EFI",
+				Default:     "BIOS",
+			},
```
Comment (VoyTechnology) [inline virtualbox/resource_vm.go:88]: If there are only two possible options, perhaps we should add a validator.

File: virtualbox/resource_vm.go [hunk 2]
```diff
@@ -80,6 +80,13 @@ func resourceVM() *schema.Resource {
 				Default:  "512mib",
 			},
 
+			"firmware": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Firmware of the vm, can either be BIOS or EFI",
```
Comment (VoyTechnology) [inline virtualbox/resource_vm.go:86]: What happens if the user provides the values in lower case, like `bios` or `efi`?

**Comment by ghouscht** [comment]
> The changes you added look good, but what is missing is adding the ability to read the firmware back in ` resourceVMRead` function.
> 
> ```go
> if err = d.Set("firmware", vm.Firmware); err != nil {
> 	return diag.Errorf("can't set firmware: %v", err)
> }
> ```

Uh, good catch, I missed that, sorry. Also done with the latest changes.

**Comment by ghouscht** [comment]
HI @VoyTechnology,
Any chance to get this merged in the near future?
Thanks, Thomas

**Pull request closed without merging**